cmd/antrea-cni/main.go

```diff
@@ -29,7 +29,7 @@ func main() {
 		cni.ActionAdd.Request,
 		cni.ActionCheck.Request,
 		cni.ActionDel.Request,
-		cni_version.PluginSupports("0.1.0", "0.2.0", "0.3.0", "0.3.1"),
+		cni_version.All,
```

Comment:
should we actually switch to `All` or just add `0.4.0` explicitly to the list of supported CNI versions? What is a new CNI version is released?

Reply:
Thinking about this more, I think my point was mostly moot. If a new CNI version is released, the value of `All` would not change until we decide to explicitly update the CNI library version we use for Antrea (https://github.com/vmware-tanzu/antrea/blob/master/go.mod#L15). And it stands to reason that we would only do that if we were ready to support the new CNI spec version...

Reply:
so in the go.mod you have cni 0.7.1 https://github.com/vmware-tanzu/antrea/blob/master/go.mod#L15 which support cni spec 0.4.0 see https://github.com/containernetworking/cni/blob/v0.7.1/pkg/version/version.go#L39 . 
So IMO the "All" should be good. 

Also I think you also need to change this https://github.com/vmware-tanzu/antrea/blob/master/pkg/agent/cniserver/server.go#L107 to be taken from the containernetworking/cni/. The support version should be defined  by cni version (/containernetworking/cni) in the go.mod. 

In the current implementation you need to change it in sevral places

Reply:
do you mind making the change to https://github.com/vmware-tanzu/antrea/blob/master/pkg/agent/cniserver/server.go#L107 as part of this PR? And any other related change you deem necessary?

Reply:
Sure will do